lib/compiler/src/v3_kernel.erl

```diff
@@ -88,6 +88,7 @@
 
 -include("core_parse.hrl").
 -include("v3_kernel.hrl").
+-define(SQUEEZE_MAX_BYTES_SEGMENT, 4).
```

Comment:
So today we squeeze at most 4 bytes together. This has two issues according to what @bjorng told me:

1. `bs_get _integer_32` is not available and is inefficient on 32 bits machines

2. This also allows us to generate a binary of 3 bytes, and we don't have an operation called `bs_get_integer_24`

I can think of two options:

1. Reduce this value to 2. The downside of reducing it to 2 is that it is less efficient for constructs of 4 bytes

3. Keep this value as 4, introduce `bs_get_integer_24` and introduce a `bs_get_integer_32` for 32 bits systems that emulates it without generating a big number. I'm unsure if this is possible though

Reply:
Matching out a 32-bit number on 32-bit systems **must** build a bignum if the signed value does not fit in a small. Introducing a `bs_get_integer_32` instruction would only be a slight improvement (avoiding a function call). There will be an extra cost building the bignum and also when comparing it.

What about reducing the value to 3? Even without a specialized instruction for extracting 24 bits, it might still be faster if the overall number of instructions are decreased. (And if it turns out to be common, we could add an `bs_get_integer_24` instruction later.)

Reply:
So in this case, I think the best option is to keep the value as 4 and rewrite 4 byte lookups into two 2-byte lookup. Otherwise we will break a 4-byte lookup into a 3byte+1byte lookup, which is not efficient.

Reply:
Btw, I added bs_integer_24 and forced max bytes lookup to be 3 and I couldn't see a meaningful difference in my benchmarks either. So I think max_size of 4 and then we explicitly break it into 2 is the best approach (I have pushed it).